ingestion/src/metadata/interfaces/datalake/datalake_profiler_interface.py

```diff
@@ -0,0 +1,198 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Interfaces with database for all database engine
+supporting sqlalchemy abstraction layer
+"""
+
+import traceback
+from collections import defaultdict
+from datetime import datetime, timezone
+from typing import Dict
+
+from pydantic import BaseModel
+from sqlalchemy import Column
+
+from metadata.generated.schema.entity.data.table import DataType, TableData
+from metadata.generated.schema.entity.services.connections.database.datalakeConnection import (
+    GCSConfig,
+    S3Config,
+)
+from metadata.ingestion.api.processor import ProfilerProcessorStatus
+from metadata.ingestion.source.database.datalake import DatalakeSource
+from metadata.interfaces.profiler_protocol import (
+    ProfilerInterfaceArgs,
+    ProfilerProtocol,
+)
+from metadata.orm_profiler.metrics.datalake_metrics_computation_registry import (
+    compute_metrics_registry,
+)
+from metadata.orm_profiler.metrics.registry import Metrics
+from metadata.orm_profiler.profiler.sampler import Sampler
+from metadata.utils.connections import get_connection
+from metadata.utils.logger import profiler_interface_registry_logger
+
+logger = profiler_interface_registry_logger()
+
+
+class DataLakeProfilerInterface(ProfilerProtocol):
+    """
+    Interface to interact with registry supporting
+    sqlalchemy.
+    """
+
+    def __init__(self, profiler_interface_args: ProfilerInterfaceArgs):
+        """Instantiate SQA Interface object"""
+        self._thread_count = profiler_interface_args.thread_count
+        self.table_entity = profiler_interface_args.table_entity
+        self.ometa_client = profiler_interface_args.ometa_client
+        self.service_connection_config = (
+            profiler_interface_args.service_connection_config
+        )
+        self.client = get_connection(self.service_connection_config).client
+        self.processor_status = ProfilerProcessorStatus()
+        self.processor_status.entity = (
+            self.table_entity.fullyQualifiedName.__root__
+            if self.table_entity.fullyQualifiedName
+            else None
+        )
+
+        self.profile_sample = profiler_interface_args.table_sample_precentage
+        self.profile_query = profiler_interface_args.table_sample_query
+        self.partition_details = None
+        self._table = profiler_interface_args.table_entity
+        self.data_frame_list = None
```

Comment:
let's call self.fetch_dl_data_frame(...) here instead. In self.fetch_dl_data_frame(...) let's just return that list